### PR TITLE
coap: allow CoAP packets to have a payload

### PR DESCRIPTION
**Pull request by todfreed**
conforming with rfc 7252. Also sets the content_format field from a Content-Format option, if any. This allows subsequent layers to bind based based on the content format value.

### CODE REVIEW

File: scapy/contrib/coap.uts [hunk 1]
```diff
@@ -47,3 +47,15 @@ s = '\x16\x33\xa0\xa4\x00\x78\xfe\x8b\x60\x45\xd9\xe1\xc1\x28\xff\x3c\x2f\x3e\x3
...
+p = CoAP(ver = 1, type = 0, code = 0x42, msg_id = 0xface, options=[("Content-Format", "\x00")], paymark = "\xff")
+p /= Raw("\xde\xad\xbe\xef")
+str(p) == '\x40\x42\xfa\xce\xc1\x00\xff\xde\xad\xbe\xef'
+
+= dissection with a text/plain payload
+p = CoAP(str(p))
+p.ver == 1
+p.type == 0
+p.code == 0x42
+p.msg_id == 0xface
```
Comment (p-l-) [inline scapy/contrib/coap.uts]: Could you add a test to check that `isinstance(p.payload, Raw)` and `p.payload.load == "\xde\xad\xbe\xef"`?

File: scapy/contrib/coap.uts [hunk 2]
```diff
@@ -47,3 +47,15 @@ s = '\x16\x33\xa0\xa4\x00\x78\xfe\x8b\x60\x45\xd9\xe1\xc1\x28\xff\x3c\x2f\x3e\x3
...
 CoAP in UDP(s)
+
+= building with a text/plain payload
+p = CoAP(ver = 1, type = 0, code = 0x42, msg_id = 0xface, options=[("Content-Format", "\x00")], paymark = "\xff")
+p /= Raw("\xde\xad\xbe\xef")
+str(p) == '\x40\x42\xfa\xce\xc1\x00\xff\xde\xad\xbe\xef'
+
+= dissection with a text/plain payload
+p = CoAP(str(p))
+p.ver == 1
```
Comment (p-l-) [inline scapy/contrib/coap.uts]: Be careful, only the last line of a test is checked. You should use `assert` statements.

**Comment by hossbeast** [comment]
accidentally closed - reopened as https://github.com/secdev/scapy/pull/549

File: scapy/contrib/coap.uts (hunk 2 above)
Comment (hossbeast) [inline scapy/contrib/coap.uts]: addressed this in 549

File: scapy/contrib/coap.uts (hunk 1 above)
Comment (hossbeast) [inline scapy/contrib/coap.uts]: addressed this in 549

**Pull request closed without merging**